src/components/App.tsx

```diff
@@ -90,15 +86,21 @@ class App extends React.Component<RouteProps, State> {
         this.setState({
           isLoading: false,
           networkErrorOccurred: true,
-          errorMessage
+          errorMessage,
+          logEntries: []
         });
       });
-  }
 
-  private handleAddEntry = (entry: LiftLogEntry) =>
-    this.liftLogService
-      .addEntry(this.logName, entry)
-      .then(() => this.reloadLifts());
+  private async handleAddEntry(entry: LiftLogEntry) {
+    this.setState(prevState => ({
+      isLoading: true,
+      loadingMessage: `Adding entry for ${entry.name}...`,
+      logEntries: prevState.logEntries
+    }));
+
+    await this.liftLogService.addEntry(this.logName, entry);
```

Comment:
why no `try/catch` around this call?

Reply:
Good point - do you think the try/catch should be around both calls?
In other words, if adding the entry fails for some reason, should we try to reload the entries?

Reply:
I think that you shouldn't try to reload entries if adding the entry failed, so it is not the same `try/catch`.